File: medicare-backend/src/main/java/com/medicare/backend/services/AdminService.java
package com.medicare.backend.services;

import java.util.List;
import com.medicare.backend.model.Medicine;


public interface AdminService {

	// Add   medicine
	public Medicine createMedicine(Medicine medicine) throws Exception;
	//Update medicine
	public Medicine updateMedicine(int medid,Medicine medicine);
	//Remove medicine
	public void removeMedicine(int medid);
	//View all medicines;
	public List<Medicine> getAllMedicines();
	//Get medicine by id
	public Medicine getMedicineByID(int medid);
	//Get medicine by name
	public Medicine getMedicineByName(String medname);
	
}

File: medicare-frontend/src/app/services/admin.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';


@Injectable({
  providedIn: 'root'
})


export class AdminService {

public currentMed:any = null;

  medicine={
    medid:'',
    medname:'',
    price:'',
    enabled:true,
    category:'',
    quantity:'',
    dosage:'',
    description:''
  };

  constructor(private http:HttpClient) { }

  public addMedicine(medicine:any){
     return this.http.post(`${baseUrl}/admin/add-medicine`,medicine);
  }
 
  public retrieveAllMedicines(){
    return this.http.get(`${baseUrl}/admin/getmedicines`);
  }

  public getMedicineByID(id:number){
    if(id == undefined){
      console.log("ANG[USER SERVICES -- ERROR --]  (getMedicineByID --> "+id+")  is UNDEFINED -- This Should Not Happen --");
    }else{
      console.log("ANG[USER SERVICES]  (getMedicineByID --> "+id+")  begin new get");
    }
    return this.http.get(`${baseUrl}/user/getmedicine/${id}`);
  }

  public updateMedicine(medid:number,medicine:any){
    return this.http.post(`${baseUrl}/admin/update-medicine/${medid}`,medicine);
  }

  public removeMedicine(medid:number){
     return this.http.delete(`${baseUrl}/admin/remove-medicine/${medid}`);
  }

}
 